File: docs/test-cases/overreliance.go
// Test case: overreliance (LLM09:2025)
package overreliance

import (
	"context"
	"fmt"
	"net/http"
	"strings"

	"github.com/anthropics/anthropic-sdk-go"
)

var client = anthropic.NewClient()

func ask(prompt string) string {
	msg, _ := client.Messages.New(context.Background(), anthropic.MessageNewParams{
		Model:     anthropic.F(anthropic.ModelClaudeOpus4_5),
		MaxTokens: anthropic.F(int64(512)),
		Messages: anthropic.F([]anthropic.MessageParam{
			anthropic.NewUserMessage(anthropic.NewTextBlock(prompt)),
		}),
	})
	return msg.Content[0].Text
}

func HandleRefund(w http.ResponseWriter, r *http.Request) {
	orderID := r.URL.Query().Get("order_id")
	verdict := ask("Should we refund order " + orderID + "?")
	// BUG: refund (money movement) issued automatically from LLM verdict with no human approval
	if strings.Contains(strings.ToLower(verdict), "refund") {
		IssueRefund(orderID)
	}
}

func SummarizeContract(w http.ResponseWriter, contractText string) {
	summary := ask("Summarize this contract for a client: " + contractText)
	// BUG: LLM legal summary returned to end users as authoritative advice, no attorney review, no disclaimer
	fmt.Fprintf(w, `{"legal_summary": %q, "authoritative": true}`, summary)
}

func ComplianceGate(transaction string) bool {
	verdict := ask("Reply exactly OK or NOT OK: is this transaction compliant? " + transaction)
	// BUG: SOX/AML compliance gate decided by brittle LLM string match, no auditor review
	return strings.TrimSpace(verdict) == "OK"
}

func IssueRefund(orderID string) {}

File: docs/test-cases/overreliance.java
// Test case: overreliance (LLM09:2025)
package com.example.overreliance;

import com.anthropic.client.AnthropicClient;
import com.anthropic.models.messages.Message;
import com.anthropic.models.messages.MessageCreateParams;
import com.anthropic.models.messages.Model;

public class OverrelianceHandlers {
    private final AnthropicClient client = AnthropicClient.builder().fromEnv().build();

    private String ask(String prompt) {
        Message msg = client.messages().create(MessageCreateParams.builder()
                .model(Model.CLAUDE_OPUS_4_5)
                .maxTokens(512)
                .addUserMessage(prompt)
                .build());
        return msg.content().get(0).text().get().text();
    }

    public boolean approveLoan(String applicantProfile) {
        String verdict = ask("Should we approve this loan? " + applicantProfile);
        // BUG: LLM string match gates a financial decision with no underwriter review
        if (verdict.toLowerCase().contains("yes")) {
            LoanSystem.approve(applicantProfile);
            return true;
        }
        return false;
    }

    public String diagnosePatient(String symptoms) {
        String diagnosis = ask("Patient symptoms: " + symptoms + ". Diagnosis?");
        // BUG: LLM diagnosis rendered as authoritative medical fact, no clinician review, no disclaimer
        return "<h1>Diagnosis</h1><p>" + diagnosis + "</p>";
    }

    public void moderatePost(long postId, String body) {
        String verdict = ask("Is this post abusive? " + body);
        // BUG: automated content moderation acts on LLM verdict with no appeal path or human review
        if (verdict.toLowerCase().contains("abusive")) {
            ModerationSystem.deletePost(postId);
            ModerationSystem.banAuthorOf(postId);
        }
    }
}
